gax-java/gax/src/test/java/com/google/api/gax/batching/BatcherImplTest.java

```diff
@@ -885,20 +888,35 @@ public void run() {
       } while (batcherAddThreadHolder.isEmpty()
           || batcherAddThreadHolder.get(0).getState() != Thread.State.WAITING);
 
+      long beforeGetCall = System.currentTimeMillis();
       executor.submit(
           () -> {
             try {
               Thread.sleep(throttledTime);
+              logger.info("Calling flowController.release");
               flowController.release(1, 1);
+              logger.info("Called flowController.release");
             } catch (InterruptedException e) {
             }
           });
 
       try {
+        logger.info("Calling future.get(10 ms)");
```

Comment:
I'm not too familiar with Batcher or this test, but what is the significance of 10ms here? If a TimeoutException is expected, why can't we lower the future.get() block time?

Reply:
From the log (in the pull request description), the problem is that the TimeoutException isn't always thrown. Lowering the 10 ms to smaller number won't help.

Reply:
> Is this just potentially a scheduling issue on a machine that has a single CPU/ few CPUs?

Yes.

Reply:
Ok, from my understanding it would be either case:
- future.get() takes more than 10ms to return -> TimeoutException
- future.get() returns a result within 10ms -> Check that the actual waitTime was at least 10ms

I think that should suffice due to the variance in scheduling. CC: @blakeli0 Do you know of another option to handle this flakiness?

Reply:
Ah, updated the comment above. The docs for `Future.get()` seem to suggest that it will throw the exception once 10ms passes, but I'm guessing that is just best effort then.